nocasedict/_nocasedict.py

```diff
@@ -37,85 +37,86 @@
 
 
 class NocaseDict(object):
+    # pylint: disable=line-too-long
     """
-    Yet another implementation of a case-insensitive dictionary.
+    A case-insensitive, ordered, and case-preserving, dictionary.
 
-    Whenever keys are looked up, that is done case-insensitively. Whenever
-    keys are returned, they are returned with the lexical case that was
-    originally specified. The dictionary preserves the order of its items
-    (on all supported Python versions).
+    The dictionary is case-insensitive: Whenever items of the dictionary are
+    looked up by key or key values are compared, that is done
+    case-insensitively. The case-insensitivity is defined by performing the
+    lookup or comparison on the result of the ``lower()`` method on the key
+    value. Therefore, objects used as keys must support the ``lower()`` method.
+    If a key object does not do that, :exc:`py:TypeError` is raised.
 
-    Except for the case-insensitivity of its keys, it behaves like the built-in
-    :class:`~py:collections.OrderedDict`.
+    The dictionary is ordered: The dictionary maintains its order based on when
+    the items have been added, for all Python versions supported by this
```

Comment:
The comma here is not required and may be semantically wrong.  Second.  should we chen "when thing have been added" to "the order in which items were added"

Reply:
DONE